libs/community/langchain_community/document_loaders/parsers/images.py

```diff
@@ -0,0 +1,313 @@
+    ):
+        """Initialize the TesseractBlobParser.
+
+        Args:
+            format (Literal["text", "markdown-img", "html-img"]):
+              The format for the parsed output.
+              - "text" = return the content as is
+              - "markdown-img" = wrap the content into an image markdown link, w/ link
+              pointing to (`![body)(#)`]
+              - "html-img" = wrap the content as the `alt` text of an tag and link to
+              (`<img alt="{body}" src="#"/>`)
+            langs (list[str]):
+              The languages to use for OCR.
+        """
+        super().__init__(format=format)
+        self.langs = list(langs)
+
+    def _analyze_image(self, img: "Image", format: str) -> str:
+        """Analyze an image and extracts text using Tesseract OCR.
+
+        Args:
+            img: The image to be analyzed.
+            format: The format to use if it's possible
+
+        Returns:
+            str: The extracted text content.
+        """
+        try:
+            import pytesseract
+        except ImportError:
+            raise ImportError(
+                "`pytesseract` package not found, please install it with "
+                "`pip install pytesseract`"
+            )
+        return pytesseract.image_to_string(img, lang="+".join(self.langs)).strip()
+
+
+_PROMPT_IMAGES_TO_DESCRIPTION: BasePromptTemplate = PromptTemplate.from_template(
+    "You are an assistant tasked with summarizing images for retrieval. "
+    "1. These summaries will be embedded and used to retrieve the raw image. "
+    "Give a concise summary of the image that is well optimized for retrieval\n"
+    "2. extract all the text from the image. "
+    "Do not exclude any content from the page.\n"
+    "Format answer in {format} without explanatory text "
+    "and without markdown delimiter ``` at the beginning. "
+    "Respects the start of the format."
+)
+
+
+class LLMImageBlobParser(BaseImageBlobParser):
+    """Parser for analyzing images using a language model (LLM).
+
+    Attributes:
+        format (Literal["text", "markdown-img", "html-img"]):
+          The format for the parsed output.
+          - "text" = return the content as is
+          - "markdown-img" = wrap the content into an image markdown link, w/ link
+          pointing to (`![body)(#)`]
+          - "html-img" = wrap the content as the `alt` text of an tag and link to
+          (`<img alt="{body}" src="#"/>`)
+          - "markdown" = return markdown content
+          - "html" = return html content
+        model (BaseChatModel):
+          The language model to use for analysis.
+        prompt (str):
+          The prompt to provide to the language model.
+    """
+
+    def __init__(
+        self,
+        *,
+        format: Literal[
+            "text", "markdown-img", "html-img", "markdown", "html"
+        ] = "text",
+        model: BaseChatModel,
+        prompt: BasePromptTemplate = _PROMPT_IMAGES_TO_DESCRIPTION,
```

Comment:
IMO just keep the original prompt that you had -- this prompt will almost certainly need to be modified, and if people are already doing that they'll also modify the format regardless

Reply:
I agree